### DISCUSSION TITLE
Adding a persistent file to an app that doesn't have anything persisted (ant)

### DISCUSSION THREAD

**Discussion by dstcruz** [ScoopInstaller/Scoop#4973] [Help and Questions] [answered]
I'm needing to add `ant-contrib.jar` to the `ant/lib` folder. However, that folder is part off `scoop/apps/ant/<version>/lib`. I've tried to read up on how to persist files across updates, but I'm not sure how to proceed here given that I don't see anything in `scoop/persist` for `ant`. What are my options for adding this `ant-contrib.jar` file to `scoop/apps/ant/<version>/lib`? Preferably so that it sticks around between updates?

**Comment by rashil2000** [comment]
Since this file doesn't always exist, you'll have to do what we call a 'manual persist'. See https://github.com/ScoopInstaller/Extras/blob/1b0deb0c39e1068b8ed101d4238217268d77d4ec/bucket/xnresize.json for example.

**Reply by dstcruz** [reply]
@rashil2000 not sure I follow. Are you suggesting that I create a (personal) manifest for `ant` that has a `pre_install` step that copies this file over? If so, I guess I need to figure out how to maintain my own manifest tree somehow? I searched the web for "manual persist" without much luck :)

**Reply by rashil2000** [reply]
You can contribute your (improved) manifest in the Main bucket by making a PR

**Comment by dstcruz** [comment, marked as answer]
This is now possible by adding files into `$persist_dir/lib` for `ant`. Also, if files are added into `$dir/lib` they will be persisted before doing an update. See https://github.com/ScoopInstaller/Main/issues/3662 for reference.
